File: stardroid-v1/app/src/test/java/com/google/android/stardroid/pushnav/PushNavConnectionProbeTest.java
/*
 * Copyright (c) 2026 Jasper Luetkens.
 *
 * This program is free software: you can redistribute it and/or modify it
 * under the terms of the GNU General Public License as published by the Free
 * Software Foundation, either version 3 of the License, or (at your option)
 * any later version.
 */
package com.google.android.stardroid.pushnav;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertThrows;

import org.junit.Test;

public class PushNavConnectionProbeTest {
  @Test
  public void normalizeServerUrl_trimsWhitespaceAndPath() {
    assertEquals(
        "http://192.168.178.50:8765",
        PushNavConnectionProbe.normalizeServerUrl(
            "  http://192.168.178.50:8765/api/version  "));
  }

  @Test
  public void normalizeServerUrl_acceptsHostnameWithoutPort() {
    assertEquals(
        "http://pushnav.local",
        PushNavConnectionProbe.normalizeServerUrl("http://pushnav.local"));
  }

  @Test
  public void normalizeServerUrl_rejectsHttps() {
    assertThrows(
        IllegalArgumentException.class,
        () -> PushNavConnectionProbe.normalizeServerUrl("https://pushnav.local:8765"));
  }

  @Test
  public void normalizeServerUrl_rejectsBlankValue() {
    assertThrows(
        IllegalArgumentException.class,
        () -> PushNavConnectionProbe.normalizeServerUrl("   "));
  }
}

File: stardroid-v1/app/src/test/java/com/google/android/stardroid/pushnav/PushNavConnectionProbeTest.kt
package com.google.android.stardroid.pushnav

import org.junit.Assert.assertEquals
import org.junit.Assert.assertThrows
import org.junit.Test

class PushNavConnectionProbeTest {
  @Test
  fun normalizeServerUrl_trimsWhitespaceAndPath() {
    assertEquals(
      "http://192.168.178.50:8765",
      PushNavConnectionProbe.normalizeServerUrl(
        "  http://192.168.178.50:8765/api/version  "
      )
    )
  }

  @Test
  fun normalizeServerUrl_acceptsHostnameWithoutPort() {
    assertEquals(
      "http://pushnav.local",
      PushNavConnectionProbe.normalizeServerUrl("http://pushnav.local")
    )
  }

  @Test
  fun normalizeServerUrl_rejectsHttps() {
    assertThrows(IllegalArgumentException::class.java) {
      PushNavConnectionProbe.normalizeServerUrl("https://pushnav.local:8765")
    }
  }

  @Test
  fun normalizeServerUrl_rejectsBlankValue() {
    assertThrows(IllegalArgumentException::class.java) {
      PushNavConnectionProbe.normalizeServerUrl("   ")
    }
  }
}
